src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2812,11 +2856,21 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                     allPathsString += fullPath;
                 }
 
-                _core.PasteText(winrt::hstring{ allPathsString });
+                _pasteTextWithBroadcast(winrt::hstring{ allPathsString });
             }
         }
     }
 
+    void TermControl::_pasteTextWithBroadcast(const winrt::hstring& text)

```

Comment:
FYI technically you can use `winrt::param::hstring` here for 0 copy string passing, but I've heard it's considered an internal API so we can't use it. 😟 (Or can we? Technically we are in control over our code... Oh well...)